cpp/net/connection_pool.cc

```diff
@@ -29,21 +145,59 @@ const HostPortPair& ConnectionPool::Connection::other_end() const {
 }
 
 
+void ConnectionPool::Connection::ReleaseConnection() {
+  conn_.release();
+  // Do not null out ssl_, despite the fact that it's a dangling pointer now we
+  // need it in the d'tor since the value of the pointer itself is a key into a
+  // map which needs to be updated.
```

Comment:
Well, you'd only erase if you have a `conn_`. And if you do, then you should `CHECK`. But the whole centralizing business might not work out so well.